src/pkg/packager/deploy.go

```diff
@@ -44,14 +44,29 @@ func (p *Packager) resetRegistryHPA() {
 // Deploy attempts to deploy the given PackageConfig.
 func (p *Packager) Deploy() (err error) {
 
-	filter := filters.Combine(
+	isInteractive := !config.CommonOptions.Confirm
+
+	deployFilter := filters.Combine(
 		filters.ByLocalOS(runtime.GOOS),
-		filters.ForDeploy(p.cfg.PkgOpts.OptionalComponents, !config.CommonOptions.Confirm),
+		filters.ForDeploy(p.cfg.PkgOpts.OptionalComponents, isInteractive),
 	)
 
-	p.cfg.Pkg, p.warnings, err = p.source.LoadPackage(p.layout, filter, true)
-	if err != nil {
-		return fmt.Errorf("unable to load the package: %w", err)
+	if isInteractive {
+		filter := filters.Empty()
```

Comment:
I am fine with this, but just wanted to note we show k3s now on mac os (still won't prompt which is what matters more IMO)

Reply:
(_if_ they read it it should be obvious that it is only linux)